# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/workers/WorkerPrivate.cpp
# Commit: 2596059bf325
# Full Hash: 2596059bf3254077a58f6bdda57796b3b20e00a5
# Author: Eden Chuang <echuang@mozilla.com>
# Date: 2025-03-18 21:32:34
# Regressor Bug: 1899503
# File Overlap Count: 2
# Description:
#   Bug 1899503 - P1 New top-level IPC PRemoteWorkerDebuggerManager and PRemoteWorkerDebugger to support nsIWorkerDebugger interface for Workers in the parent process. r=asuth
#   
#   New IPC PRemoteWorkerDebuggerManager and PRemoteWorkerDebugger are introduced to support remote debugger of Workers on the parent thread.
#   
#   PRemoteWorkerDebuggerManager is IPC between the parent process main thread and the content process WorkerLauncher thread.
# ==============================================================================

diff -r 2cd7a5fae8ef -r 2596059bf325 dom/workers/WorkerPrivate.cpp
--- a/dom/workers/WorkerPrivate.cpp	Tue Mar 18 11:37:28 2025 +0000
+++ b/dom/workers/WorkerPrivate.cpp	Tue Mar 18 12:10:53 2025 +0000
@@ -6463,7 +6463,7 @@
 }
 
 const nsAString& WorkerPrivate::Id() {
-  AssertIsOnMainThread();
+  AssertIsOnParentThread();
 
   if (mId.IsEmpty()) {
     mId = ComputeWorkerPrivateId();